### PR TITLE
Remove if else

### PR DESCRIPTION
**Pull request by pavan-pan**
I have removed the complex if/else branch. I have used an array instead to add the scores of different classes. Finding the highest score is much easier and faster with an array.

### CODE REVIEW

File: functions.c [hunk 1]
```diff
@@ -321,6 +321,90 @@ int show_weather(char* weather) {
     return 0;
 }
 
+bool check_date_validity (int yy, int mm, int dd)
```
Comment (ritwik12) [inline functions.c]: @pavan-pan Are you adding reminder functionality also in this PR?

**Comment by ritwik12** [comment]
@pavan-pan Thanks a lot for your PR. There are a lot of changes in this PR. Please give me sometime to review it. It looks good.

File: functions.c (hunk 1 above)
Comment (pavan-pan) [reply functions.c]: I am half way into the reminder but please ignore it for now.

Comment (ritwik12) [reply functions.c]: I can ignore but Git won't :) 
You should create new branches for each PR so that there isn't a conflict.

File: main.c [hunk 2]
```diff
@@ -5,9 +5,9 @@
 #include <curl/curl.h>
 #include <json-c/json_object.h>
 #include <json-c/json_tokener.h>
+#include <stdbool.h>
```
Comment (riya-17) [inline main.c]: @pavan-pan What is the use of this header file in this program?

Comment (pavan-pan) [reply main.c]: @riya-17  This header file has the definition for bool datatype

**Comment by ritwik12** [comment]
@pavan-pan Is this WIP? if not, can you please clean this PR so that I can check it.

File: main.c (hunk 2 above)
Comment (riya-17) [reply main.c]: @pavan-pan Ok, Thanks for the PR :)

**Comment by pavan-pan** [comment]
All the changes concerning reminder are removed

**Comment by ritwik12** [comment]
@pavan-pan I'm out of town right now and my laptop isn't with me. Sorry I need to hold this for few days. I will merge it once I'll come back and run it on my system. Hope that's fine.

**Comment by ritwik12** [comment]
@pavan-pan I've tested this but it isn't giving any outputs. Any input results in No output at all. Please check if it's working in your system.

**Comment by pavan-pan** [comment]
@ritwik12 I just checked it seems to be working on my system. There is no output when there are blanks and spaces otherwise it results in output.

**Comment by ritwik12** [comment]
@pavan-pan Compile command is same i.e `gcc main.c $(pkg-config --libs --cflags libcurl) -l json-c -std=gnu11`, right?

**Comment by ritwik12** [comment]
@pavan-pan it is not working for me.

**Comment by pavan-pan** [comment]
@ritwik12 let me check again

**Comment by pavan-pan** [comment]
@ritwik12 I checked out the branch pavan-pan:remove_if_else and compiled with the same which you mentioned above. It works fine for me. 

I tried with commands weather, restaurants, google etc. I am not able to guess what could be the problem.

**Comment by pavan-pan** [comment]
@ritwik12 I have tested it on 3 different computers. I am getting the results whenever I enter something.

File: requests.c [hunk 3]
```diff
@@ -5,11 +5,26 @@
  * Created on April 10, 2018, 8:51 AM
  */
 
+bool only_spaces = true;
```
Comment (ritwik12) [inline requests.c]: You need to pull this branch with the master branch as these changes have already been merged.

**Comment by ritwik12** [comment]
Have you downloaded from https://github.com/pavan-pan/Virtual-Assistant/archive/remove_if_else.zip and then tried running it? I guess you have done some changes in your system code which isn't here. I am still not able to run it. It isn't giving any outputs. I have gone through the code and everything looks fine.

**Comment by ritwik12** [comment]
@riya-17 @speedy-software-ch @andy5995 Can you please check if this is working in your system? Maybe I am doing something wrong. I am running it on Fedora 28.

**Comment by pavan-pan** [comment]
@ritwik12 I downloaded again from the link you mentioned above. Typed "weather" and got the weather for my current location.

**Comment by pavan-pan** [comment]
I ran git pull and its updated 
capture

**Comment by ritwik12** [comment]
@pavan-pan It's updated with your forked master, but in the upstream we have already merged "space" functionality.

**Comment by pavan-pan** [comment]
@ritwik12 can you try it now ?

**Comment by ritwik12** [comment]
@pavan-pan Tried, still same. After downloading the repo, Did you changed anything, anywhere?

**Comment by pavan-pan** [comment]
@ritwik12 I change nothing. Just download, run the command that you mentioned above, run the executable, nothing more.

**Comment by ritwik12** [comment]
@pavan-pan Which linux version?

**Comment by pavan-pan** [comment]
I am running on linux mint. I am about to try it on arch to try on a different environment.

**Comment by ritwik12** [comment]
No new dependencies right?

**Comment by pavan-pan** [comment]
nope. Just download, compile and run. I just downloaded and ran on arch linux. No problems there. I really dont understand why. Are we running the same code ?

**Comment by ritwik12** [comment]
@pavan-pan Yes, I am using this branch's code https://github.com/pavan-pan/Virtual-Assistant/tree/remove_if_else

**Comment by ritwik12** [comment]
@pavan-pan Please send me the screenshot of compile and run. Maybe I am doing something wrong.

**Comment by pavan-pan** [comment]
@ritwik12 
capture1
capture2

**Comment by ritwik12** [comment]
@pavan-pan It looks good, let's wait for others to try it. I will also keep on checking and try to run it.

**Comment by riya-17** [comment]
Hey @pavan-pan, your code is working fine but only youtube and Calendar functionality are executing, for other commands it is not performing any action and just asking "Hey, How can I help you?" again and again.

**Comment by ritwik12** [comment]
> Hey @pavan-pan, your code is working fine but only youtube and Calendar functionality are executing, for other commands it is not performing any action and just asking "Hey, How can I help you?" again and again.

@pavan-pan Seconded

**Comment by pavan-pan** [comment]
@ritwik12 @riya-17 lemme check it again

**Comment by pavan-pan** [comment]
@ritwik12 @riya-17 the code is still working for me correctly. I am not sure what the problem is. Is it possible to send me the code in a zip in a mail or some way ?

**Comment by riya-17** [comment]
@pavan-pan Yaa sure we can try that. Please mail the code on these email ids : riyad17031998@gmail.com and ritwik.1204@gmail.com . We will look into it and get back to you. 
Thanks :)

**Comment by pavan-pan** [comment]
@riya-17 I have sent you the code

**Comment by pavan-pan** [comment]
Any one checked the code i sent ?

**Comment by ritwik12** [comment]
> Any one checked the code i sent ?

Yes, we did and it was giving the same outputs as we have told you. There's some issue I guess.

**Comment by pavan-pan** [comment]
@ritwik12 if you can give me more information about where its failing or under what conditions its failing i can try to solve this problem. The reason i need this information is because i cannot reproduce the issue.

**Comment by ritwik12** [comment]
@pavan-pan Thanks a lot for all your hardwork here, I guess we couldn't reach to a conclusion here. Please feel free to contribute in future too.
Closing this now

**Pull request closed without merging**